compiler/simplCore/SimplMonad.hs

```diff
@@ -55,7 +58,7 @@ newtype SimplM result
                 -> UniqSupply   -- We thread the unique supply because
                                 -- constantly splitting it is rather expensive
                 -> SimplCount
-                -> IO (result, UniqSupply, SimplCount)}
+                -> IO (result, [Id], UniqSupply, SimplCount)}
```

Comment:
Is the `[Id]` just here for debugging then? Or do you intend to keep it?

In the former case, indicate with a `TODO: mp: remove the [Id] when the simplifier is stabilised`

In the latter, indicate with a `TODO: implement the writer [Id] as a state monad for efficiency`